packages/bierzo-wallet/src/routes/welcome/index.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { ConnectedRouter } from 'connected-react-router';
+import { mount, ReactWrapper } from 'enzyme';
+import { createMemoryHistory } from 'history';
+import MedulasThemeProvider from 'medulas-react-components/lib/theme/MedulasThemeProvider';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { configureStore } from '../../store';
+import { createRootReducer } from '../../store/reducers';
+import { globalStyles } from '../../theme/globalStyles';
+import { Welcome } from '../welcome';
+
+describe('The welcome route', () => {
+  let fakeHistory, fakeReducer, fakeStore, wrapper: ReactWrapper, button: ReactWrapper;
+
+  beforeEach(() => {
+    fakeHistory = createMemoryHistory();
+    fakeReducer = createRootReducer(fakeHistory);
+    fakeStore = configureStore(fakeReducer, fakeHistory);
+
+    wrapper = mount(
```

Comment:
Maybe because you are using enzyme?

Reply:
Enzyme was the way that I found to mount the component and surround it with the necessary components to pass the Store, the History, and the ThemeProvider.

I believe the current problem is that Jest does not correctly parse the imported MedulasThemeProvider, so it fails. When I checked that the tests where correctly implemented was when I did not use the ThemeProvider component.

However, if I remove the ThemeProvider now, it also fails because the makeStyles function of the components expect a "theme" variable that before didn't use to expect (when I was not using theming in the CSS).

Reply:
I will look into it. Using enzyme is something which makes sense, of course, is a good library and well supported, but React updates like in the 18.3 version makes some problems into them. Let's wait for @albertandrejev, I want to hear his opinion as well, but my gut tells me to use the react test utilities done by react itself. We have used it in the past and is enough.

Reply:
To be honest we have three possible options to do testing:
1. Snapshot using storybook (I would strongly recommend to prepare stories for storybook for all used views in webapp);
2. We have DOM tests using React TestUtils;
3. We have E2E tests using puppeteer.
So I think adding additional option for testing will be too redundantly and not required. We should be consistent in all parts of application. Also we have examples for all cases in chrome extensions package.

Reply:
@abefernan left a request changes comment with steps to follow. Once they are done let me know and I will specify the next ones.